CONTRIBUTING.md

```diff
@@ -418,191 +351,1122 @@ This is called the `virtual path`. In this case it should be `stocks.dps.shorted
+
+TO BE ADDED - WORK IN PROGRESS
+
+## Write Code and Commit
+
+At this stage it is assumed that you have already forked the project and are ready to start working.
+
+### Pre Commit Hooks
+
+Git hook scripts are useful for identifying simple issues before submission to code review. We run our hooks on every
+commit to automatically point out issues in code such as missing semicolons, trailing whitespace, and debug statements.
+By pointing these issues out before code review, this allows a code reviewer to focus on the architecture of a change
+while not wasting time with trivial style nitpicks.
+
+Install the pre-commit hooks by running: `pre-commit install`.
+
+### Coding
+
+Although the Coding Guidelines section has been already explained. It is worth mentioning that if you want to be faster
+at developing a new feature, you may implement it first on a `jupyter notebook` and then carry it across to the
+terminal. This is mostly useful when the feature relies on scraping data from a website, or implementing a Neural
+Network model.
+
+### Git Process
+
+1. Create your Feature Branch, e.g. `git checkout -b feature/AmazingFeature`
+2. Check the files you have touched using `git status`
+3. Stage the files you want to commit, e.g.
+   `git add openbb_terminal/stocks/stocks_controller.py openbb_terminal/stocks/stocks_helper.py`.
+   Note: **DON'T** add `config_terminal.py` or `.env` files with personal information, or even `feature_flags.py` which is user-dependent.
+4. Write a concise commit message under 50 characters, e.g. `git commit -m "meaningful commit message"`. If your PR
+   solves an issue raised by a user, you may specify such issue by adding #ISSUE_NUMBER to the commit message, so that
+   these get linked. Note: If you installed pre-commit hooks and one of the formatters re-formats your code, you'll need
+   to go back to step 3 to add these.
+
+## Add a Test
+
+Unit tests minimize errors in code and quickly find errors when they do arise.
+
+### Pytest
+
+Pytest allows users to quickly create unit tests in Python. To use pytest run `pytest tests/`.
+
+### Coverage
+
+Coverage allows users to see how complete unit tests are for Python. To use coverage do the following:
+
+1. `coverage run -m pytest`
+2. `coverage html`
+
+To view the tests find the htmlcov folder in the main directory and open the _index.html_ file. This will show a detailed
+report of testing coverage.
+
+### VCR
+
+VCRPY allows us to save data from request methods to a .YAML file. This increases test integrity and significantly
+speeds up the time it takes to run tests. To use VCRPY add **@pytest.mark.vcr** above any function you write.
+
+## OpenBB API
+  
+### Understanding API infrastructure
+
+Work in progress. For official documentation go to: <https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/openbb_terminal/API_README.md>
```

Comment:
Probably remove this if we will have an API readme, and this is about usability?

Reply:
We remove the topic? Right now yes, that document will be improved later